src/config_file.js

```diff
@@ -0,0 +1,81 @@
+/*
+ * Manage the UI configuration file.
+ */
+
+'use strict'
+
+const fs = require('fs')
+const RQ_PARAMS = require('./params.js')
+
+class ConfigFile {
+  /**
+   * Manage the persistence of the UI configuration file and its
+   * version property.
+   *
+   */
+  constructor () {
+    this._setup_config()
+  }
+
+  /**
+   * If the configuration file does not exist in the persistence
+   * directory, copy the image version of the file to the persistence
+   * directory.
+   */
+  _setup_config () {
+    if (fs.existsSync(RQ_PARAMS.CONFIG_FILE)) {
+      return true
+    } else {
+      fs.copyFileSync(
+        RQ_PARAMS.DEFAULT_CONFIG_FILE,
+        RQ_PARAMS.CONFIG_FILE
+      )
+      console.log(`config file setup ${RQ_PARAMS.CONFIG_FILE}`)
+    }
+  }
+
+  /**
+   * Save the updated configuration in the persistence directory.
+   * Any existing previous configuration file is removed. The existing
+   * configuration file is moved to the .old file. Lastly, the configuration
+   * object is stringified and written.
+   *
+   * This is a synchronous call, so it's expensive.
+   *
+   * @param {string|object} configuration - the configuration
+   */
+  save_config (configuration) {
+    const oldConfigFile = RQ_PARAMS.CONFIG_FILE + '.old'
+    if (fs.existsSync(oldConfigFile)) {
+      fs.rmSync(oldConfigFile)
+    }
+
+    fs.renameSync(RQ_PARAMS.CONFIG_FILE, oldConfigFile)
+
+    if (typeof configuration === 'string') {
+      configuration = JSON.parse(configuration)
```

Comment:
I suggest wrapping the json parse into a try catch block, send an error back on not able to parse. There are a lot of reasons why a json might not parse so it's a good spot to check and have feedback.

Reply:
Good catch.